src/Discord.Net/API/DiscordSocketApiClient.cs

```diff
@@ -190,8 +190,10 @@ internal override void Dispose(bool disposing)
                 Token = _authToken,
                 Properties = props,
                 LargeThreshold = largeThreshold,
-                UseCompression = useCompression
+                UseCompression = useCompression,
+                ShardingParams = totalShards > 1 ? new int[] { shardID, totalShards } : null
```

Comment:
See other comment about Optional<T>s, dont assign this value at all if it isn't needed.